File: Sorting Algorithms/Brick Sort/BrickSort.rb
# Brick Sort Algorithm in Ruby

def brick_sort(arr)
    n = arr.length
    sorted = false
  
    while !sorted
      sorted = true
      0.upto(n - 2) do |i|
        if arr[i] > arr[i + 1]
          arr[i], arr[i + 1] = arr[i + 1], arr[i]
          sorted = false
        end
      end
  
      (n - 2).downto(0) do |i|
        if arr[i] > arr[i + 1]
          arr[i], arr[i + 1] = arr[i + 1], arr[i]
          sorted = false
        end
      end
    end
  
    arr
  end
  
  # Example usage
  arr = [5, 2, 8, 1, 9, 3, 7, 4, 6]
  sorted_arr = brick_sort(arr)
  puts sorted_arr # Output: [1, 2, 3, 4, 5, 6, 7, 8, 9]
  

File: Sorting Algorithms/Brick Sort/BrickSort.py
def odd_even_brick_sort(arr):
    """
    Perform odd-even transposition sort (Brick Sort) on the input array.

    Parameters:
    arr (list): List to be sorted

    Returns:
    list: Sorted list
    """
    n = len(arr)
    sorted = False
    while not sorted:
        sorted = True
        for i in range(1, n - 1, 2):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                sorted = False
        for i in range(0, n - 1, 2):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                sorted = False
    return arr

# Example Usage
my_list = [64, 34, 25, 12, 22, 11, 90]
sorted_list = odd_even_brick_sort(my_list)
print(sorted_list)
